### ISSUE TITLE
Trouble with Media Queries

### ISSUE THREAD

**Issue by sansarip** [dhleong/spade#6]
I'm not sure if I'm doing something wrong or if there's a bug with the media query implementation.

Here's what I tried:

```clojure
(defclass test-class []
          (at-media {:min-width "200px"}
                    {:margin-left "10em"})
          {:background-color "blue"
           :margin-left "1em"
           :width "4em"
           :height "4em"})

(defcard
  test
  "Testing media queries"
  (fn []
    (sab/html
      [:div {:class (test-class)}])))
```

Here's what it results in:

![devcard](https://i.gyazo.com/45bb63ded4c69b6e7d9ae6fab46e4829.png)

The classes work perfectly except for media queries. I've tried all manner of things to get media queries to work, but nothing seems to take effect. 

Please advise, thanks!

**Comment by dhleong** [comment]
Media queries are currently ordered as-is, and for whatever reason the
media query does not add specificity to the things inside, so the default
"things after in the file override things before" CSS thing is happening
here.

If you move the at-media below the other map it should work. You can see
the CSS that gets generated by looking for the appropriate style element in
the head. It should have an ID based on the fully qualified var name in
debug mode.

I agree it's tempting to reorder media queries to try to handle this but I
feel like that might just be trading one source of confusion for another...

**Comment by sansarip** [comment]
Hey, thanks for the speedy reply!

So as you suggested I reordered the arguments as follows:
```clojure
(defclass test-class []
          {:background-color "blue"
           :margin-left "1em"
           :width "4em"
           :height "4em"}
          (at-media {:min-width "200px"}
                    {:margin-left "10em"}))
```

Unfortunately, that did not affect anything visually on the page.

Looking at the generated css it makes sense as to why nothing happened - the media query isn't selecting anything to apply the `margin-left` attribute to:

```css
.my-website-cards-home-test-class {
  background-color: blue;
  margin-left: 1em;
  width: 4em;
  height: 4em;
}

@media (min-width: 200px) {

  margin-left: 10em;

}
```

I tried giving the media-query the generated class name:

```clojure
(defclass test-class []
          {:background-color "blue"
           :margin-left "1em"
           :width "4em"
           :height "4em"}
          (at-media {:min-width "200px"}
                    [".my-website-cards-home-test-class"
                     {:margin-left "10em"}]))
```

But that resulted in unexpected css - you can see that the class gets selected twice:

```css
.my-website-cards-home-test-class {
  background-color: blue;
  margin-left: 1em;
  width: 4em;
  height: 4em;
}

@media (min-width: 200px) {

  .my-website-cards-home-test-class .my-website-cards-home-test-class {
    margin-left: 10em;
  }

}
```
If I manually remove one of the duplicate `.my-website-cards-home-test-class` selectors from the media query via the browser debugger, then it works as expected.

I must be going about this the wrong way or something :confounded: ! Please advise.

**Comment by dhleong** [comment]
Huh. Okay you're right there's definitely a bug. I just fired up the demo and it looks like you should be able to put `at-media` at the beginning (it does already get ordered after in the generated CSS) but is missing the class name.

Here's a workaround for now:

```clojure
(defclass test-class []
  (at-media {:min-width "200px"}
     [:& {:margin-left "10em"}])
  {:background-color "blue"
   :margin-left "1em"
   :width "4em"
   :height "4em"})
```                     

Thanks for the report by the way!

**Comment by dhleong** [comment]
Okay after looking into this, it appears that garden doesn't actually support nesting media queries inside classes just yet. See https://github.com/noprompt/garden/issues/171 for the relevant issue. I may be able to fake it for now by just automatically wrapping the body of an `at-media` in that `[:&]` syntax however. I'll play with it a bit

**Comment by dhleong** [comment]
Released v1.0.2 that fixes this issue. Thanks again :)
